application/src/main/kotlin/net/corda/v5/application/flows/ResponderFlow.kt

```diff
@@ -0,0 +1,35 @@
+package net.corda.v5.application.flows
+
+import net.corda.v5.application.messaging.FlowSession
+import net.corda.v5.base.annotations.Suspendable
+
+/**
+ * A flow that may be started by receiving a message from a peer.
+ *
+ * A flow implementing this will also need to be marked as @InitiatedBy to be invoked via a session message. If both
+ * these requirements are met, then the flow will be invoked via the `call` method that takes a session. This session is
+ * created by the platform and communicates with the party that initiated the session.
+ *
+ * Note that an initiated flow cannot return a value.
+ */
+interface ResponderFlow<out T> : Flow<T> {
+
+    /**
+     * The business logic for the flow should be written here.
+     *
+     * This is equivalent to the call method for a normal flow. This version is invoked when the flow is started via an
+     * incoming session init event, via a counterparty calling `initiateFlow`.
+     *
+     * @param session The session opened by the counterparty.
+     */
+    @Suspendable
+    fun call(session: FlowSession)
```

Comment:
Although, it makes sense that there is no return type from the flow, there is an argument that it makes it harder to test because can't verify the "output" or end result of the flow.

If we think that testing via mocks and using `verify` will be sufficient, then going with no return type is fine.

If we keep the return type, then users can still define it as `Unit` if they don't want to return anything, however Java  can't really do that and has to still return `null` which is annoying.

Reply:
If we use the code you have at the moment, you should probably remove the generic type off of `ResponderFlow` and just have `Flow<Unit>`.

Reply:
I'd rather have the API make sense from a user perspective and work around any testing problems than put something on the API that's misleading to assist our testing. The issue with giving a return type is the user can't do anything with it, we just swallow it.